strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/LdapAuthenticatorConfigurationController.java

```diff
@@ -0,0 +1,352 @@
+            return ResponseEntity.badRequest().body(LdapMessages.NOT_CONFIGURED);
+        }
+        try
+        {
+            String previousInternalRole = springSecurityLdapInternalsSupplier.getAuthoritiesMapper().addRoleMapping(
+                    externalRole, internalRole);
+            if (previousInternalRole != null)
+            {
+                return ResponseEntity.badRequest().body("LDAP role mapping already exists for given LDAP role");
+            }
+        }
+        catch (Exception e)
+        {
+            return toError("Addition of LDAP role mapping failed with server error: " + e.getLocalizedMessage());
+        }
+
+        return ResponseEntity.ok("LDAP role mapping addition succeeded");
+    }
+
+    @ApiOperation(value = "Adds or updates LDAP role mapping")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "LDAP role mapping add or update succeeded"),
+                            @ApiResponse(code = 400, message = "LDAP is not enabled"),
+                            @ApiResponse(code = 500, message = "Insert or update LDAP role mapping failed with server error") })
+    @RequestMapping(value = "/rolesMapping/{externalRole}/{internalRole}", method = RequestMethod.PUT)
+    public ResponseEntity<String> setRoleMapping(@PathVariable String externalRole,
+                                                 @PathVariable String internalRole)
+    {
+        if (!springSecurityLdapInternalsSupplier.isLdapAuthenticationEnabled())
+        {
+            return ResponseEntity.badRequest().body(LdapMessages.NOT_CONFIGURED);
+        }
+        try
+        {
+            springSecurityLdapInternalsSupplier.getAuthoritiesMapper().upsertRoleMapping(externalRole, internalRole);
+        }
+        catch (Exception e)
+        {
+            return toError("Add or update LDAP role mapping failed with server error: " + e.getLocalizedMessage());
+        }
+
+        return ResponseEntity.ok("LDAP role mapping add or update succeeded");
+    }
+
+
+    @ApiOperation(value = "Deletes LDAP role mapping")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "LDAP role mapping deletion succeeded"),
+                            @ApiResponse(code = 400, message = "LDAP is not enabled or externalRole does not exist in the LDAP roles mapping"),
+                            @ApiResponse(code = 500, message = "LDAP role mapping deletion failed with server error") })
+    @RequestMapping(value = "/rolesMapping/{externalRole}", method = RequestMethod.DELETE)
+    public ResponseEntity<String> deleteRoleMapping(@PathVariable String externalRole)
+    {
+        if (!springSecurityLdapInternalsSupplier.isLdapAuthenticationEnabled())
+        {
+            return ResponseEntity.badRequest().body(LdapMessages.NOT_CONFIGURED);
+        }
+        try
+        {
+            String removedInternalRole = springSecurityLdapInternalsSupplier.getAuthoritiesMapper().deleteRoleMapping(
+                    externalRole);
```

Comment:
Cosmetic: I would normally indent the chained method by the first method like this:
```
String removedInternalRole = springSecurityLdapInternalsSupplier.getAuthoritiesMapper()
                                                                .deleteRoleMapping(externalRole);